File: src/main/java/cad97/spawnercraft/proxy/ClientProxy.java
package cad97.spawnercraft.proxy;

import cad97.spawnercraft.init.SpawnerCraftBlocks;
import cad97.spawnercraft.init.SpawnerCraftItems;
import net.minecraft.client.Minecraft;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

public final class ClientProxy extends CommonProxy {
    @Override
    public void preInit(FMLPreInitializationEvent event) {
        super.preInit(event);
        SpawnerCraftBlocks.registerModels();
        SpawnerCraftItems.registerModels();
    }

    @Override
    public void init(FMLInitializationEvent event) {
        super.init(event);
        SpawnerCraftItems.registerColors(Minecraft.getMinecraft().getItemColors());
    }

}

File: src/main/kotlin/cad97/spawnercraftkt/proxy/ClientProxy.kt
package cad97.spawnercraftkt.proxy

import cad97.spawnercraftkt.init.SpawnerCraftBlocks
import cad97.spawnercraftkt.init.SpawnerCraftItems
import net.minecraft.client.Minecraft
import net.minecraftforge.fml.common.event.FMLInitializationEvent

class ClientProxy : Proxy() {
    companion object {
        // ClientProxy::class.java.qualifiedName
        const val qualifiedName = "cad97.spawnercraftkt.proxy.ClientProxy"
    }

    override fun init(event: FMLInitializationEvent) {
        super.init(event)
        SpawnerCraftItems.registerColors(Minecraft.getMinecraft().itemColors)
        SpawnerCraftBlocks.registerColors(Minecraft.getMinecraft().itemColors, Minecraft.getMinecraft().blockColors)
    }
}
